test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/BulkUpdateSaveResultTests.cs

```diff
@@ -0,0 +1,288 @@
+﻿//---------------------------------------------------------------------

```

Comment:
I think you should create more test cases to validate more scenarios and edge cases:
- scenarios where some, but not all, properties in entities were changed
- scenarios where multiple entities have a link to the same entity, both on the same or different navigation properties
- scenarios with more than 2 levels of nesting
- scenarios with cycles in the relationship graph